src/core/data.jl

```diff
@@ -3,348 +3,453 @@
+
+Return a dictionary of the active component indices.
+A component is inactive if its status value is 1.
+
+```
+    julia> get_active_items(network)
+    Dict{String, Set{String}} with 4 entries:
+        "gen"     => Set(["4", "1", "2"])
+        "branch"  => Set(["4", "1", "5", "2", "6", "7", "3"])
+        "storage" => Set(["1"])
+        "bus"     => Set(["4"])
+```
+"""
+function get_active_items(network::Dict{String, Any})
+    pm_data = _PM.get_pm_data(network)
+    if _IM.ismultinetwork(pm_data)
+        Memento.error(_PM._LOGGER, "get_active_items can only be used on single networks")
+    else
+        active_items = Dict{String, Set{String}}()
+        for comp_type in restoration_comps
+            status_key = _PM.pm_component_status[comp_type]
+            active_items[comp_type] = Set()
+            for (comp_id, comp) in get(network, comp_type, Dict())
+                if haskey(comp, status_key) && comp[status_key] != PowerModels.pm_component_status_inactive[comp_type]
+                    push!(active_items[comp_type], comp_id)
+                end
             end
         end
+        return active_items
     end
 end
 
 
-"Replace NaN and Nothing with 0 in multinetwork solutions"
-function clean_solution!(solution::Dict{String,Any})
-    for item_type in ["gen", "storage", "branch","load","shunt"]
-        if haskey(solution["solution"], "nw")
-            for (n, net) in solution["solution"]["nw"]
-                for (i,item) in get(net, item_type, Dict())
-                    for k in keys(item)
-                        if item[k] === nothing
-                            item[k] = 0
-                        elseif isnan(item[k])
-                            item[k] = 0
-                        end
-                    end
-                end
-            end
-        else
-            for (i,item) in get(solution["solution"], item_type, Dict())
-                for k in keys(item)
-                    if item[k] === nothing
-                        item[k] = 0
-                    elseif isnan(item[k])
-                        item[k] = 0
-                    end
-                end
+"""
+    count_active_items(network::Dict{String, <:Any})
+
+Count the number of items with an active component status.
+"""
+function count_active_items(network::Dict{String, Any})
+    pm_data = _PM.get_pm_data(network)
+    if _IM.ismultinetwork(pm_data)
+        Memento.error(_PM._LOGGER, "count_active_items can only be used on single networks")
+    else
+        active_set = get_active_items(pm_data)
+        return sum(length(comp_ids) for (comp_name,comp_ids) in active_set)
+    end
+end
+
+
+"""
+    set_comp_inactive!(nw_data::Dict{String,<:Any}, comp_list::Dict{String, Set{String}})
+
+Set the status indicator to 0 for components in the comp_list.
+
+```
+    julia> set_comp_inactive!(network, Dict("bus"=>["1","3"]))
+```
+"""
+function set_comp_inactive!(network::Dict{String,<:Any}, comp_list::Dict{String, Set{String}})
```

Comment:
I am wondering if `set_inactive!` or `make_inactive!` would be more intuitive?

Reply:
I like `make_inactive!`